xpk.py

```diff
@@ -3231,6 +3332,65 @@ def create_cluster_if_necessary(
     )
 
 
+def is_cluster_using_clouddns(args) -> int:
+  """Checks if cluster is using CloudDNS.
+  Args:
+    args: user provided arguments for running the command.
+
+  Returns:
+    0 if cluster is using CloudDNS and 1 otherwise.
+  """
+  command = (
+      f'gcloud container clusters describe {args.cluster}'
+      f' --project={args.project} --region={zone_to_region(args.zone)}'
+      ' | grep "clusterDns: CLOUD_DNS" | wc -l'
+  )
+  return_code, cloud_dns_matches = run_command_for_value(
+      command,
+      'Check if Cloud DNS is enabled in cluster describe.',
+      args,
+  )
+  if return_code != 0:
+    xpk_exit(return_code)
+  cloud_dns_matches = int(cloud_dns_matches)
+  if cloud_dns_matches > 0:
+    xpk_print('Cloud DNS is enabled on the cluster, no update needed.')
+    return 0
+  return 1
```

Comment:
Returning values the other way round seem more intuitive (1 means true/cluster is using clouddns and 0 otherwise)

Reply:
Could return a bool instead? true / false. ints are great for error checking vs this looks like a good use of bool logic

Reply:
I agree, I followed the convention followed in all of the XPK functions where 0 is successful and 1 means error.

Reply:
Thanks, sure, can switch to boolean without issues!